experimental/LieAlgebras/src/root_systems.jl

```diff
@@ -0,0 +1,164 @@
+mutable struct RootSystem <: AbstractVector{fmpq}
+  roots::Vector{Vector{Int}}
+  simple_roots::Vector{Vector{Int}}
+  positive_roots::Vector{Vector{Int}}
+  root_system_type::Vector{Union{String, Int64}}
+  @doc raw"""
+    `RootSystem(S::String) -> RootSystem`
+  Return the root system of type `S` where `S` is a string consisting out of
+  a letter `A`, `B`, `C`, `D`, `E`, `F`, `G` followed by an integer.
+  For the exceptional root system, the integers are fixed, e.g. `G2`, `F4`, `E6`, ...
+  """
+  function RootSystem(S::String)
+    # S is a string detailing the type of the indecomposable root system made out of a letter, e.g. "A", "B", "C",... and an integer for the number of simple roots
+    l = length(S)
+    n = parse(Int64, S[2:l])
+    S1 = GAP.Obj(string(S[1]))
```

Comment:
You could also match what you do a few lines later:
```suggestion
    S1 = GAP.Obj(S[1:1])
```